arcgis-android-toolkit/src/main/java/com/esri/arcgisruntime/toolkit/ar/ArcGISArView.kt

```diff
@@ -330,58 +506,80 @@ final class ArcGISArView : FrameLayout, DefaultLifecycleObserver, Scene.OnUpdate
      * @since 100.6.0
      */
     @SuppressLint("MissingPermission") // suppressed as function returns if permission hasn't been granted
-    private fun beginSession() {
+    fun startTracking() {
         initializationStatus = ArcGISArViewState.INITIALIZING
 
-        try {
-            (context as? Activity)?.let {
-                // ARCore requires camera permissions to operate. If we did not yet obtain runtime
-                // permission on Android M and above, now is a good time to ask the user for it.
-                // when the permission is requested and the user responds to the request from the OS this is executed again
-                // during onResume()
-                if (!hasPermission(CAMERA_PERMISSION)) {
+        checkArCoreAvailability()
+
+        (context as? Activity)?.let { activity ->
+            // ARCore requires camera permissions to operate. If we did not yet obtain runtime
+            // permission on Android M and above, now is a good time to ask the user for it.
+            // when the permission is requested and the user responds to the request from the OS this is executed again
+            // during onResume()
+            if (isUsingARCore == ARCoreUsage.YES && renderVideoFeed && !hasPermission(CAMERA_PERMISSION)) {
+                requestPermission(
+                    activity,
+                    CAMERA_PERMISSION,
+                    CAMERA_PERMISSION_CODE
+                )
+                return
+            }
+
+            // Request location permission if user has provided a LocationDataSource
+            locationDataSource?.let {
+                if (!hasPermission(LOCATION_PERMISSION)) {
                     requestPermission(
-                        it,
-                        CAMERA_PERMISSION,
-                        CAMERA_PERMISSION_CODE
+                        activity,
+                        LOCATION_PERMISSION,
+                        LOCATION_PERMISSION_CODE
                     )
                     return
                 }
```

Comment:
Just wondering if we could request camera and location permissions in one go if required. It would make for a nicer user experience rather than prompting twice.

Reply:
We don't need the location permission if we aren't using a location data source and we don't need the camera position if we aren't using ARCore so doesn't make sense to request both at the same time.

Reply:
So using ARCore and using a location data source are mutual exclusive? In other words, we either use ARCore or location data source, but never both at the same time?

Reply:
We can use both at the same time, I'll re-work to request both at the same time if necessary.